spring-cloud-stream-binders/spring-cloud-stream-binder-kafka/src/test/java/org/springframework/cloud/stream/binder/kafka/KafkaBinderTests.java

```diff
@@ -282,16 +284,15 @@ public void testCustomPartitionCountDoesNotOverridePartitioningIfSmaller() throw
 
 		byte[] ratherBigPayload = new byte[2048];
 		Arrays.fill(ratherBigPayload, (byte) 65);
-		KafkaTestBinder binder = (KafkaTestBinder) getBinder();
+		KafkaTestBinder binder = getBinder();
 
 		DirectChannel moduleOutputChannel = new DirectChannel();
 		QueueChannel moduleInputChannel = new QueueChannel();
-		Properties producerProperties = new Properties();
-		producerProperties.put(BinderPropertyKeys.MIN_PARTITION_COUNT, "3");
-		producerProperties.put(BinderPropertyKeys.NEXT_MODULE_COUNT, "5");
-		producerProperties.put(BinderPropertyKeys.PARTITION_KEY_EXPRESSION, "payload");
-		Properties consumerProperties = new Properties();
-		consumerProperties.put(BinderPropertyKeys.MIN_PARTITION_COUNT, "3");
+		KafkaProducerProperties producerProperties = new KafkaProducerProperties();
+		producerProperties.setPartitionCount(5);
```

Comment:
No, the old property just didn't make sense anymore. The partition count used to be computed from the next module count and the next module concurrency, so instead of that we just use the partition count. The property was an override, which doesn't make sense. If a producer can set the partition count, they might as well set it right.